src/main/java/org/folio/services/budget/BudgetService.java

```diff
@@ -146,26 +153,30 @@ private List<Error> checkRemainingEncumbrance(SharedBudget budget) {
   private List<Error> checkRemainingExpenditure(SharedBudget budget) {
     log.debug("checkRemainingExpenditure:: Check remaining expenditure for budget: {}", budget.getId());
     BigDecimal allocated = BigDecimal.valueOf(budget.getAllocated());
+    BigDecimal netTransfers = BigDecimal.valueOf(budget.getNetTransfers());
     BigDecimal expenditures = BigDecimal.valueOf(budget.getExpenditures());
     BigDecimal credits = BigDecimal.valueOf(budget.getCredits());
     BigDecimal awaitingPayment = BigDecimal.valueOf(budget.getAwaitingPayment());
-    BigDecimal available = BigDecimal.valueOf(budget.getAvailable());
-    BigDecimal unavailable = BigDecimal.valueOf(budget.getUnavailable());
 
-    //[amount we can expend] = (allocated * allowableExpenditure) - (allocated - (unavailable + available)) -
-    // (expended - credited + awaitingPayment)
+    // [remaining amount we can expend] = (allocated + netTransfers) * allowableExpenditure - (awaitingPayment + expended - credited)
+    // calculations should be consistent with BatchTransactionChecks in mod-finance-storage
     if (budget.getAllowableExpenditure() != null) {
       log.info("checkRemainingExpenditure:: Budget '{}' allowable expenditure is not null", budget.getId());
-      BigDecimal newAllowableExpenditure = BigDecimal.valueOf(budget.getAllowableExpenditure())
-        .movePointLeft(2);
-      if (allocated.multiply(newAllowableExpenditure)
-        .subtract(allocated.subtract(available.add(unavailable)))
-        .subtract(expenditures.subtract(credits).add(awaitingPayment))
-        .compareTo(BigDecimal.ZERO) < 0) {
+      BigDecimal allowableExpenditure = BigDecimal.valueOf(budget.getAllowableExpenditure()).movePointLeft(2);
+      BigDecimal totalFunding  = allocated.add(netTransfers);
+      // unavailable amount shouldn't be negative
+      BigDecimal unavailable = ensureNonNegative(awaitingPayment.add(expenditures).subtract(credits));
+
+      double remaining = totalFunding.multiply(allowableExpenditure).subtract(unavailable).doubleValue();
+      if (remaining < 0) {
```

Comment:
Maybe better to take without `doubleValue` and compareTo(ZERO)

Reply:
I thought about this, but decided to remain as in mod-finance-storage, to have exactly the same consistent code

https://github.com/folio-org/mod-finance-storage/blob/master/src/main/java/org/folio/service/transactions/batch/BatchTransactionChecks.java#L215

https://github.com/folio-org/mod-finance-storage/blob/master/src/main/java/org/folio/service/transactions/batch/BatchTransactionChecks.java#L248